content/blog/2023-12-26-xmpp-providers-fully-automated.md

```diff
@@ -0,0 +1,75 @@
+
+## API v2
+
+After automating much of the work which needed to be done manually previously, some provider properties did not seem fitting anymore and needed to change.
+This alone is of course no reason for bumping an API's version.
+While automating the process, additional properties were added, because they were available through the tools.
+
+#### Which properties did change in API `v2`?
+
+- `lastCheck` has been replaced by `latestUpdate`, since checks now run daily.
+- `company` has been replaced by `organization`, which allows for a finer distinction of an organization's type.
+
+#### Additional new properties that are collected automatically
+
+- Account creation process: For determining whether an email address is required for registering an account, [XEP-0077: In-Band Registration](https://xmpp.org/extensions/xep-0077.html) is used.
+- Account creation process: For determining whether a [CAPTCHA](https://en.wikipedia.org/wiki/CAPTCHA) need to be solved for registering an account, [XEP-0077: In-Band Registration](https://xmpp.org/extensions/xep-0077.html) is used.
+- Support Addresses: For retrieving the support addresses of a provider, [XEP-0157: Contact Addresses for XMPP Services](https://xmpp.org/extensions/xep-0157.html) is used. Please note that due to limitations of the standard, only one language is supported for addresses.
+
+The [FAQ section](/faq/#where-do-we-have-the-providers-properties-from) explains how these properties can be provided by server admins.
+
+**Additional new properties:**
+
+- `alternativeJids`: A list of JIDs a provider offers for registration other than the main JID.
+- `serverTesting`: Whether tests against the provider's server are allowed (e.g., certificate checks and uptime monitoring).
+
+## A Provider File For More Automation
+
+Not all of the properties collected by the XMPP Providers project are machine-readable yet.
+To enable automatic collection of the missing properties, the team works on extending existing standards and, if necessary, creating new ones.
+
+As a workaround until standards have been extended or created, and until those changes have been implemented and deployed in the wild, a 'provider file' shall fill the gap.
+A provider file is a JSON file containing only the provider properties that cannot be retrieved via other methods. Each provider can generate a provider file and supply it via its web server.
+To make this as easy as possible, a [Provider File Generator](/provider-file-generator/) has been developed, which generates a provider file from the information you supply.
+
+Example URL for a provider file of version 2:
+
+`https://example.org/.well-known/xmpp-provider-v2.json`
+
+As soon as a provider file is discovered by the tools, all properties listed in the provider file are automatically fetched and processed.
+
+#### Properties which can be fetched automatically via a provider file
```

Comment:
```suggestion
#### Changed Properties
```

Reply:
That's a different topic you're suggesting for that chapter. This chapter is about the provider file primarily, not changed properties.